From 903e717e1200ecb720a66a52ce20e6bf538b3811 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Thu, 27 Sep 2018 14:27:11 +0700
Subject: [PATCH 211/557] mmc: renesas: renesas_sdhi_core: Change bus_shift to
 support iwg23s

Fix MMC timeout waiting for SD bus idle.

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/mmc/host/renesas_sdhi_core.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/renesas_sdhi_core.c b/drivers/mmc/host/renesas_sdhi_core.c
index 2f8c7b8..cda8e50 100644
--- a/drivers/mmc/host/renesas_sdhi_core.c
+++ b/drivers/mmc/host/renesas_sdhi_core.c
@@ -523,9 +523,14 @@ int renesas_sdhi_probe(struct platform_device *pdev,
 			renesas_sdhi_start_signal_voltage_switch;
 	}
 
-	/* Orginally registers were 16 bit apart, could be 32 or 64 nowadays */
-	if (!host->bus_shift && resource_size(res) > 0x100) /* old way to determine the shift */
-		host->bus_shift = 1;
+	/* SD control register space size */
+	if (resource_size(res) > 0x400) /* 0x400 for bus_shift=2 */
+		host->bus_shift = 2;
+ 	/* SD control register space size is 0x100, 0x200 for bus_shift=1 */
+	else if (resource_size(res) > 0x100)
+ 		host->bus_shift = 1;
+ 	else
+ 		host->bus_shift = 0;
 
 	if (mmd)
 		*mmc_data = *mmd;
-- 
2.7.4

